lib/inspec/control_eval_context.rb

```diff
@@ -137,8 +142,8 @@ def to_s
         end
 
         # method for attributes; import attribute handling
-        define_method :attribute do |name, options|
-          profile_context_owner.register_attribute(name, options)
+        define_method :attribute do |name, options = {}|
+          profile_context_owner.find_or_create_attribute(name, options)
```

Comment:
Same concern as above; in a test block, I'd imagine that you only want to access existing attributes.  If you're able to autovivify attributes, that will hide typos, etc.